heat/nn/data_parallel.py

```diff
@@ -70,6 +80,10 @@ def __init__(
         optimizer: Union[optim.DataParallelOptimizer, List, Tuple],
         blocking_parameter_updates: bool = False,
     ):
+        if isinstance(optimizer, optim.DASO):
+            raise TypeError(
+                "For use with DASO please use DataParallelMultiGPU instead of DataParallel"
```

Comment:
i think that would take too much work to unify them. they use vastly different methods of communication. So i think its easier to just throw the error. and its also at the top of the code so that helps

Reply:
agreed